sunpy/image/transform.py

```diff
@@ -167,7 +107,175 @@ def _get_transform_method(method, use_scipy):
+#       The translational shift to apply to the image in each axis
+#   order : `int`
+#       The numerical parameter that controls the degree of interpolation
+#   missing : `float`
+#       The value to use for outside the bounds of the original image, and may be NaN
+#   clip : `bool`
+#       Whether to clip the output image to the range of the input image
+#
+# There are three decorators that can be used:
+#
+# * `_handle_clipping_externally` is for rotation functions that do not implement clipping
+# * `_handle_nan_missing_externally` is for rotation functions that do not handle NaN for `missing`
+# * `_handle_image_nans_externally` is for rotation functions that do not handle NaNs in the image
+#
+# Do not modify `image` directly; copy first if necessary.
+
+
+@_handle_clipping_externally
+@_handle_image_nans_externally
+def _rotation_scipy(image, matrix, shift, order, missing, clip):
+    """
+    Rotation using SciPy
+    """
+    rotated_image = scipy.ndimage.affine_transform(image.T, matrix, offset=shift, order=order,
+                                                   mode='constant', cval=missing).T
+
+    return rotated_image
+
+
+@_handle_clipping_externally
+@_handle_nan_missing_externally
+@_handle_image_nans_externally
+def _rotation_skimage(image, matrix, shift, order, missing, clip):
+    """
+    Rotation using scikit-image
+
+    We handle clipping externally because the clipping behavior of
+    :func:`skimage.transform.warp` is different between orders in {0, 1, 3} and
+    orders in {2, 4, 5}.
+
+    We handle NaN for `missing` externally because :func:`skimage.transform.warp`
+    does not handle it correctly for orders in {2, 4, 5}.
+    """
+    import skimage.transform
+
+    # Make the rotation matrix 3x3 to include translation of the image
+    skmatrix = np.zeros((3, 3))
+    skmatrix[:2, :2] = matrix
+    skmatrix[2, 2] = 1.0
+    skmatrix[:2, 2] = shift
+    tform = skimage.transform.AffineTransform(skmatrix)
+
+    if issubclass(image.dtype.type, numbers.Integral):
+        warn_user("Integer input data has been cast to float64.")
+        adjusted_image = image.astype(np.float64)
+    else:
+        adjusted_image = image.copy()
+
+    # Scale image to range [0, 1]
+    im_min = np.nanmin([missing, np.min(adjusted_image)])
+    adjusted_image -= im_min
+    adjusted_missing = missing - im_min
+    im_max = np.nanmax([adjusted_missing, np.max(adjusted_image)])
+    if im_max > 0:
+        adjusted_image /= im_max
+        adjusted_missing /= im_max
+
+    rotated_image = skimage.transform.warp(adjusted_image, tform, order=order,
+                                           mode='constant', cval=adjusted_missing, clip=clip)
+
+    # Convert the image back to its original range
+    if im_max > 0:
+        rotated_image *= im_max
+    rotated_image += im_min
+
+    return rotated_image
+
+
+_supported_rotation_methods = {'scipy': _rotation_scipy,
+                               'skimage': _rotation_skimage}
```

Comment:
Rather than having this dict be manually populated, you could have a decorator build it automatically? :D